### PR TITLE
updating dependency logic for bayesian-changepoint-detection

### PR DESCRIPTION
**Pull request by sayanchk**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-git+https://github.com/hildensia/bayesian_changepoint_detection@2dd95f5c1d028116899a842ccb3baa173f9d5be9#egg=bayesian-changepoint-detection
+bayesian-changepoint-detection @ git+https://github.com/hildensia/bayesian_changepoint_detection@2dd95f5c1d028116899a842ccb3baa173f9d5be9#egg=bayesian-changepoint-detection
```
Comment (shahsmit14) [inline requirements.txt:1]: B: My assumption was we are also going to remove it from the requirements.txt file since that gets used inside setup.py file.

**Comment by shahsmit14** [comment]
B: Also, the pipeline is failing. That needs to be fixed.

File: requirements.txt (hunk 1 above)
Comment (sayanchk) [reply requirements.txt:1]: The build is failing. I added this back to test if it is due to this

**Comment by shahsmit14** [comment]
We would need to squash all your commits into one before merging since its just adding lot of commit noise.

**Comment by sayanchk** [comment]
Build is failing due to hyperopt incompatibility with python 3.10 due to it's dependency on py4j (the recent py4j 0.10.9.4 release is throwing errors in python 3.10). I have added a temporary py4j dependency <= 0.10.9.3 until the issue is fixed on hyperopt / py4j end.

**Review by shahsmit14 — APPROVED** [review]
Make sure to squash your commits.

**Pull request merged**